providers/azureblob/src/main/java/org/jclouds/azureblob/config/AzureBlobHttpApiModule.java

```diff
@@ -63,6 +64,17 @@ protected final String guiceProvideTimeStamp(@TimeStamp Supplier<String> cache)
    protected String provideTimeStamp(@TimeStamp Supplier<String> cache) {
       return cache.get();
    }
+   
+   /** 
+   * checks which Authentication type is used 
+   *
+   */
+   @Named("sasAuth")
+   @Provides 
+   protected Boolean authSAS(@org.jclouds.location.Provider Supplier<Credentials> creds) {
+	  String credential =  creds.get().credential;
+      return credential.matches("sv.*se.*sig.*");
```

Comment:
So, would the following be a valid SAS credential? `svsesig`

I think we need some unit tests here to make it clear what are valid SAS formats, and also a stronger regular expression. Do the SAS credential has delimiters separating each field? Can we enforce they start with `sv`?, etc.

Reply:
So, I did some investigation. sv token may appear at the beginning, in the middle or at the end of the string, as well as any other token. 
There are two types of SAS (Account and Service SAS), they both work with the code.
There are optional tokens and required token. If we imagine their tokens as two sets, there is an intersection, which is sv, se, st, sig and sp. But there are many other tokens, which may or may not appear in Account or Service SAS. And the position of the token doesn't matter at all. The only regex, I can now come up with, is a very long one. Take a union of the both sets, and separate them with OR operator. 
There are also patterns - for example, after se there will always be a timestamp. To make matters worse, the timestamp may also differ. And so on. 
I would think of splitting the SAS String by '&' and then apply .match(regex). 
But the regex would be a very long one.

What would you think of such an approach?

Reply:
With this regexp, for example, tokens where the `se` token comes before the `sv` one won't match, and those are valid tokens, right?
I'd suggest you start writing a bunch of unit tests with several (many) different combinations of valid and invalid SAS tokens, then build a regexp that properly honors the contract defined by the tests.

Reply:
yes, with this one `svsesig` the ones with se coming first won't work. 
I have thought of something like this: 
` boolean authSAS(Supplier<Credentials> creds) {
		String testSAS = creds.get().credential;
		String[] sasPieces = testSAS.split(testSAS);	
		String regex = "(sv=[0-9]{4}-(0[1-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1]))|(sr=(b|c))|(sp=[rwdlacup]{1,8})|(se=.{10,22})|(sig=.{45,55})|(sip=.*)|(spr=(https,http)|https)|(rscc.*)|(rscd.*)|(rsce.*)|(rscl.*)|(rsct.*)|(si=.*)|(api-version=.*)|(ss=[bqtf]{1,4})|(srt=[sco]{1,3}).*";
		
		for(String sasPiece : sasPieces) {
			if(!sasPiece.matches(regex)) {
				return false;
			}
		}
		return true;
	   }`

Reply:
On the other hand, the SAS is just a query string. There would probably be a way to check if it is a query string  with the use of java URI classes. And we are not able to check its validity anyway, so here it would be more about format and how to differ Key and SAS.

Reply:
But in any case I need to write unit tests

Reply:
If that is a query map it may be better and easier to parse it and just check for the presence of all mandatory keys. Something like:
```java
import static com.google.common.base.Predicates.in;
import static com.google.common.base.Predicates.not;
import static com.google.common.collect.Iterables.all;

@Named("sasAuth")
@Provides
protected Boolean authSAS(@org.jclouds.location.Provider Supplier<Credentials> creds) {
   String credential = creds.get().credential;
   List<String> required = ImmutableList.of("sv", "se", "st", "sig", "sp");
   try {
      Map<String, String> tokens = Splitter.on('&').withKeyValueSeparator('=').split(credential);
      return all(required, in(tokens.keySet()));
   } catch (Exception ex) {
      return false;
   }
}
```

Reply:
Yes, I find this idea suitable. I have just implemented and tested it, everything works as expected. 
Pushed the change, thanks!